index.bs

```diff
@@ -3341,6 +3341,219 @@ be the same.
+      void stroke(Path2D path);
+    };
+</pre>
+
+Per the ECMAScript language binding, calling <code>stroke(undefined)</code> on a object implementing
+{{CanvasDrawPath}} would attempt to call the second overload, yielding a {{TypeError}} since
+<emu-val>undefined</emu-val> cannot be <a href="#es-to-interface">converted</a> to a {{Path2D}}.
+However, if the operations were instead defined with [=optional arguments=] and merged into one,
+
+<pre highlight="webidl">
+    interface CanvasDrawPathExcerpt {
+      void stroke(optional Path2D path);
+    };
+</pre>
+
+the [=overload resolution algorithm=] would treat the <var ignore>path</var> argument as not present
+given the same ECMAScript language code <code>stroke(undefined)</code>, and not throw any
+exceptions.
+
+Additionally, there are semantic differences as well. [=Overloaded=] operations are designed to map
+well to language features such as C++ overloading, and are usually a better fit for operations with
+more substantial differences in what they do given arguments of different types. [=Union types=], in
+contrast, are usually used in the sense that "any of the types would work in about the same way".
+
+That being said, we offer the following recommendations and examples in case of difficulties to
+determine what Web IDL language feature to use:
+
+*   When the operation must return values of different types for different argument types,
+    [=overloaded|overloading=] is almost always a better choice.
+
+    Suppose there is an operation <code class="idl">calculateWithType()</code> that accepts a single
+    argument of either {{long}}, {{DOMString}}, or <code class="idl">CalculatableInterface</code>
+    (an interface type) types, and returns a value of the same type as its argument. It would be
+    clearer to write the IDL fragment using [=overloaded=] operations as
+
+    <pre highlight="webidl">
+        interface A {
+          long calculateWithType(long input);
+          DOMString calculateWithType(DOMString input);
+          CalculatableInterface calculateWithType(CalculatableInterface input);
+        };
+    </pre>
+
+    than using a [=union type=] with a [=typedef=] as
+
+    <pre highlight="webidl">
+        typedef (long or DOMString or CalculatableInterface) Calculatable;
+        interface A {
+          Calculatable calculateWithType(Calculatable input);
+        };
+    </pre>
+
+    which does not convey the fact that the return value is always of the same type as <var
+    ignore>input</var>.
+
+    The problem is exacerbated when one of the overloads has a return type of {{void}}, since
+    [=union types=] cannot even contain {{void}} as a [=member type=]. In that case, a return type
+    of {{any}} must be used with appropriate prose defining the return type, further decreasing
+    expressiveness.
+
+*   When the operation has significantly different semantics for different argument types or
+    lengths, [=overloaded|overloading=] is preferred.
```

Comment:
Maybe: "Again, in such scenarios, it is usually better to create separate operations with distinct names, but legacy APIs sometimes follow this pattern."